src/libraries/System.Private.CoreLib/src/System/WeakReference.cs

```diff
@@ -17,12 +124,54 @@ public partial class WeakReference : ISerializable
         // attacks (i.e. if the WeakReference instance is finalized away underneath you when you're still
         // handling a cached value of the handle then the handle could be freed and reused).
 
-#if !CORECLR
-        // the handle field is effectively readonly until the object is finalized.
-        private IntPtr _handleAndKind;
+        private nint_t _handleAndKind;
+
+        // the lowermost 2 bits are reserved for storing additional info about the handle
+        // we can use these bits because handle is at least 32bit aligned
+        // we also reserve the sign bit
+#if CORECLR
+        private const nint_t HandleTagBits = nint_t.MinValue | 3;
```

Comment:
Same comments here about clarify on `MinValue`.

Reply:
I also think a type with constants for these values is appropriate. Having 3 distinct locations is troublesome. We should at most have a managed location and a native location.